src/Build/BackEnd/Node/OutOfProcNode.cs

```diff
@@ -503,10 +503,9 @@ private NodeEngineShutdownReason HandleShutdown(out Exception exception)
             // Shutdown any Out Of Proc Nodes Created
             _taskHostNodeManager.ShutdownConnectedNodes(_shutdownReason == NodeEngineShutdownReason.BuildCompleteReuse);
 
-#if FEATURE_ENVIRONMENT_SYSTEMDIRECTORY
-            // Restore the original current directory.
-            NativeMethodsShared.SetCurrentDirectory(Environment.SystemDirectory);
-#endif
+            // On Windows, a process holds a handle to the current directory,
+            // so reset it away from a user-requested folder that may get deleted.
+            NativeMethodsShared.SetCurrentDirectory(BuildEnvironmentHelper.Instance.CurrentMSBuildToolsDirectory);
```

Comment:
Please check my logic about the new default value for this!

I can also do an if-windows conditional.

Reply:
Can setting cwd to a bogus value have any external behaviour? Files written, node failure logs, etc? If yes, do an if-windows conditional. If no, it doesn't really matter.

Reply:
AFAIK none of that depends on CWD . . .